services/bettersocial/post/createPostV3.js

```diff
@@ -190,14 +190,19 @@ async function processSendSystemMessage(
   selfUserId,
   username,
   with_system_message = false,
-  isAnonimous = false
+  isAnonimous = false,
+  processSendSystemMessage = 'never'
 ) {
   if (!with_system_message) return;
   if (!filteredTopics) return;
   if (filteredTopics?.length == 0) return;
   if (!selfUserId || !username) throw new Error('Missing params');
 
   console.log('send system message');
+  const post_expired_at =
+    processSendSystemMessage !== 'never'
+      ? moment().utc().add(+processSendSystemMessage, 'day')
+      : moment().utc().add(100, 'years');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `processSendSystemMessage` parameter is being used as a string and a number at the same time. This can lead to confusion and potential bugs in the future. It would be better to separate these two functionalities into two different parameters, one for controlling whether the system message should be sent or not (a boolean), and another for specifying the number of days until the post expires (a number). Also, using magic numbers like '100 years' can make the code harder to understand and maintain. Consider defining this value as a constant at the top of your file.
```diff
-  processSendSystemMessage = 'never'
+  sendSystemMessage = false,
+  postExpiryDays = 36500 // Default to 100 years

...

- const post_expired_at =
-   processSendSystemMessage !== 'never'
-     ? moment().utc().add(+processSendSystemMessage, 'day')
-     : moment().utc().add(100, 'years');
+ const post_expired_at = moment().utc().add(sendSystemMessage ? postExpiryDays : 100 * 365, 'days');
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->